### PR TITLE
Report junit pytest failures to CircleCI

### PR DESCRIPTION
**Pull request by eric-wieser**
Results will be at https://app.circleci.com/pipelines/github/sphinx-doc/sphinx/824/workflows/6cd51dea-285d-4ff9-b03d-301c062eae9f/jobs/7821/tests if this works

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -70,7 +70,9 @@ pylint:
 
 .PHONY: test
 test:
-	@$(PYTHON) -m pytest -v $(TEST)
+	# for circleCI, path must match what is there
+	@mkdir -p test-reports/pytest
+	@$(PYTHON) -m pytest -v $(TEST) --junitxml=test-reports/pytest/results.xml
```
Comment (tk0miya) [inline Makefile]: It's a bit odd to write a filename `test-report/pytest` to two different files. It is better to control it all in `.circleci/config.yml`. How about this instead?
```
- run: make -p test-reports/pytest
- run: make test PYTHON=/python3.6/bin/python TEST=--junitxml=test-reports/pytest/results.xml
```

Comment (eric-wieser) [reply Makefile]: That looks sensible, thanks

**Comment by eric-wieser** [comment]
Seems to work now, updated the PR description with a link

**Comment by tk0miya** [comment]
Thanks!

**Pull request merged**